frontend/src/main/scala/bloop/engine/BuildLoader.scala

```diff
@@ -6,12 +6,39 @@ import bloop.io.AbsolutePath
 import bloop.logging.{DebugFilter, Logger}
 import bloop.io.ByteHasher
 import monix.eval.Task
+import io.circe.Json
+import bloop.data.SettingsMapper
+import bloop.config.Config
 
 object BuildLoader {
 
+  /** File to store Metals specific settings*/
+  val settingsFileName = "settings.json"
+
   /** The pattern used to find configuration files */
   private[bloop] final val JsonFilePattern: String = "glob:**.json"
 
+  /**
+   * Gets information from additional data in the initialisation request to determine the settings for SemanticDB plugin
+   *
+   * @param data       Additional data from initialise request to interpret.
+   * @param configDir  Main bloop configuration directory.
+   * @param logger     The logger that collects messages about saving the settings file.
+   */
+  def writeSettingsFile(data: Option[Json], configDir: AbsolutePath, logger: Logger) : Unit = {
```

Comment:
I propose we move this to the file where we define settings (`SettingsMapper`) so that we don't clutter this API. Also, it might be a good idea to move the logic reading/writing this file to the caller of `BuildLoader` and add a `settings: Settings` field to `Build`. Therefore, all the build loader does is only loading the build instead of handling the settings :smile: